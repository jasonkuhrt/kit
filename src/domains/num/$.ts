/**
 * Numeric types and utilities with branded types for mathematical constraints.
 *
 * Provides branded number types (Positive, Negative, Even, Odd, etc.) with
 * runtime validation, mathematical operations, range types, and specialized
 * numeric domains like Complex, Ratio, and BigInt. Includes type guards,
 * ordering, and equivalence utilities.
 */
export * as Num from './$$.js'
