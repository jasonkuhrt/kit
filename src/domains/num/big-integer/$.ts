export * as BigInteger from './$$.ts'
