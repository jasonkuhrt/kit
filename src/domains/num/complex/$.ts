export * as Complex from './$$.js'
