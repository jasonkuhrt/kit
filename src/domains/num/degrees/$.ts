// Export the namespace containing all Degrees operations
export * as Degrees from './$$.js'
