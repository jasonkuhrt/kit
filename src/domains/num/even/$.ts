// Export the namespace containing all Even operations
export * as Even from './$$.ts'
