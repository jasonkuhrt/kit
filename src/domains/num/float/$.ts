// Export the namespace containing all Float operations
export * as Float from './$$.js'
