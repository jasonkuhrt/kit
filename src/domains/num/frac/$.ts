// Export the namespace containing all Frac operations
export * as Frac from './$$.js'
