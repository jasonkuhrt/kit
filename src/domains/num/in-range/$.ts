// Export the namespace containing all InRange operations
export * as InRange from './$$.js'
