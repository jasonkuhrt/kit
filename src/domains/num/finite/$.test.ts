import { Test } from '#test'
import { Finite } from './$.js'

Test.describe('is')
  .on(Finite.is)
  .cases(
    // Valid cases - finite numbers
    [0],
    [1],
    [-1],
    [1.5],
    [-1.5],
    [Number.MAX_SAFE_INTEGER],
    [Number.MIN_SAFE_INTEGER],
    // Invalid cases - non-finite numbers
    [Infinity],
    [-Infinity],
    [NaN],
    ['0'],
    [null],
    [undefined],
  )
  .test()

Test.describe('from')
  .on(Finite.from)
  .cases(
    // Valid cases
    [0],
    [1],
    [-1],
    [1.5],
    [-1.5],
    [Number.MAX_SAFE_INTEGER],
    [Number.MIN_SAFE_INTEGER],
    // Invalid cases - should throw
    [Infinity],
    [-Infinity],
    [NaN],
  )
  .test()

Test.describe('tryFrom')
  .on(Finite.tryFrom)
  .cases(
    // Valid cases
    [0],
    [1],
    [-1],
    [1.5],
    [-1.5],
    [Number.MAX_SAFE_INTEGER],
    [Number.MIN_SAFE_INTEGER],
    // Invalid cases - should return null
    [Infinity],
    [-Infinity],
    [NaN],
  )
  .test()
