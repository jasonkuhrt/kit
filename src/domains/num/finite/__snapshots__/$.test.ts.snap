// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`from > from(-1) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
-1
╠══════════════════════════════════════════════════╣ THEN RETURNS
-1
╚══════════════════════════════════════════════════╝"
`;

exports[`from > from(-1.5) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
-1.5
╠══════════════════════════════════════════════════╣ THEN RETURNS
-1.5
╚══════════════════════════════════════════════════╝"
`;

exports[`from > from(-9007199254740991) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
-9007199254740991
╠══════════════════════════════════════════════════╣ THEN RETURNS
-9007199254740991
╚══════════════════════════════════════════════════╝"
`;

exports[`from > from(0) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
0
╠══════════════════════════════════════════════════╣ THEN RETURNS
0
╚══════════════════════════════════════════════════╝"
`;

exports[`from > from(1) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
1
╠══════════════════════════════════════════════════╣ THEN RETURNS
1
╚══════════════════════════════════════════════════╝"
`;

exports[`from > from(1.5) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
1.5
╠══════════════════════════════════════════════════╣ THEN RETURNS
1.5
╚══════════════════════════════════════════════════╝"
`;

exports[`from > from(9007199254740991) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
9007199254740991
╠══════════════════════════════════════════════════╣ THEN RETURNS
9007199254740991
╚══════════════════════════════════════════════════╝"
`;

exports[`from > from(null) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
null
╠══════════════════════════════════════════════════╣ THEN THROWS
Error: Value must be finite, got: Infinity
╚══════════════════════════════════════════════════╝"
`;

exports[`from > from(null) 2`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
null
╠══════════════════════════════════════════════════╣ THEN THROWS
Error: Value must be finite, got: -Infinity
╚══════════════════════════════════════════════════╝"
`;

exports[`from > from(null) 3`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
null
╠══════════════════════════════════════════════════╣ THEN THROWS
Error: Value must be finite, got: NaN
╚══════════════════════════════════════════════════╝"
`;

exports[`is > is("0") 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
"0"
╠══════════════════════════════════════════════════╣ THEN RETURNS
false
╚══════════════════════════════════════════════════╝"
`;

exports[`is > is(-1) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
-1
╠══════════════════════════════════════════════════╣ THEN RETURNS
true
╚══════════════════════════════════════════════════╝"
`;

exports[`is > is(-1.5) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
-1.5
╠══════════════════════════════════════════════════╣ THEN RETURNS
true
╚══════════════════════════════════════════════════╝"
`;

exports[`is > is(-9007199254740991) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
-9007199254740991
╠══════════════════════════════════════════════════╣ THEN RETURNS
true
╚══════════════════════════════════════════════════╝"
`;

exports[`is > is(0) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
0
╠══════════════════════════════════════════════════╣ THEN RETURNS
true
╚══════════════════════════════════════════════════╝"
`;

exports[`is > is(1) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
1
╠══════════════════════════════════════════════════╣ THEN RETURNS
true
╚══════════════════════════════════════════════════╝"
`;

exports[`is > is(1.5) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
1.5
╠══════════════════════════════════════════════════╣ THEN RETURNS
true
╚══════════════════════════════════════════════════╝"
`;

exports[`is > is(9007199254740991) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
9007199254740991
╠══════════════════════════════════════════════════╣ THEN RETURNS
true
╚══════════════════════════════════════════════════╝"
`;

exports[`is > is(null) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
null
╠══════════════════════════════════════════════════╣ THEN RETURNS
false
╚══════════════════════════════════════════════════╝"
`;

exports[`is > is(null) 2`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
null
╠══════════════════════════════════════════════════╣ THEN RETURNS
false
╚══════════════════════════════════════════════════╝"
`;

exports[`is > is(null) 3`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
null
╠══════════════════════════════════════════════════╣ THEN RETURNS
false
╚══════════════════════════════════════════════════╝"
`;

exports[`is > is(null) 4`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
null
╠══════════════════════════════════════════════════╣ THEN RETURNS
false
╚══════════════════════════════════════════════════╝"
`;

exports[`is > is(undefined) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
undefined
╠══════════════════════════════════════════════════╣ THEN RETURNS
false
╚══════════════════════════════════════════════════╝"
`;

exports[`tryFrom > tryFrom(-1) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
-1
╠══════════════════════════════════════════════════╣ THEN RETURNS
-1
╚══════════════════════════════════════════════════╝"
`;

exports[`tryFrom > tryFrom(-1.5) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
-1.5
╠══════════════════════════════════════════════════╣ THEN RETURNS
-1.5
╚══════════════════════════════════════════════════╝"
`;

exports[`tryFrom > tryFrom(-9007199254740991) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
-9007199254740991
╠══════════════════════════════════════════════════╣ THEN RETURNS
-9007199254740991
╚══════════════════════════════════════════════════╝"
`;

exports[`tryFrom > tryFrom(0) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
0
╠══════════════════════════════════════════════════╣ THEN RETURNS
0
╚══════════════════════════════════════════════════╝"
`;

exports[`tryFrom > tryFrom(1) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
1
╠══════════════════════════════════════════════════╣ THEN RETURNS
1
╚══════════════════════════════════════════════════╝"
`;

exports[`tryFrom > tryFrom(1.5) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
1.5
╠══════════════════════════════════════════════════╣ THEN RETURNS
1.5
╚══════════════════════════════════════════════════╝"
`;

exports[`tryFrom > tryFrom(9007199254740991) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
9007199254740991
╠══════════════════════════════════════════════════╣ THEN RETURNS
9007199254740991
╚══════════════════════════════════════════════════╝"
`;

exports[`tryFrom > tryFrom(null) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
null
╠══════════════════════════════════════════════════╣ THEN RETURNS
null
╚══════════════════════════════════════════════════╝"
`;

exports[`tryFrom > tryFrom(null) 2`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
null
╠══════════════════════════════════════════════════╣ THEN RETURNS
null
╚══════════════════════════════════════════════════╝"
`;

exports[`tryFrom > tryFrom(null) 3`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
null
╠══════════════════════════════════════════════════╣ THEN RETURNS
null
╚══════════════════════════════════════════════════╝"
`;
